app/src/main/java/de/westnordost/streetcomplete/quests/barrier_opening/AddBarrierOpening.kt

```diff
@@ -0,0 +1,79 @@
+package de.westnordost.streetcomplete.quests.barrier_opening
+
+import de.westnordost.streetcomplete.R
+import de.westnordost.streetcomplete.data.elementfilter.toElementFilterExpression
+import de.westnordost.streetcomplete.data.osm.geometry.ElementGeometry
+import de.westnordost.streetcomplete.data.osm.mapdata.Element
+import de.westnordost.streetcomplete.data.osm.mapdata.MapDataWithGeometry
+import de.westnordost.streetcomplete.data.osm.osmquests.OsmElementQuestType
+import de.westnordost.streetcomplete.data.user.achievements.EditTypeAchievement.BICYCLIST
+import de.westnordost.streetcomplete.data.user.achievements.EditTypeAchievement.CAR
+import de.westnordost.streetcomplete.data.user.achievements.EditTypeAchievement.PEDESTRIAN
+import de.westnordost.streetcomplete.data.user.achievements.EditTypeAchievement.WHEELCHAIR
+import de.westnordost.streetcomplete.osm.Tags
+import de.westnordost.streetcomplete.quests.width.WidthAnswer
+import de.westnordost.streetcomplete.screens.measure.ArSupportChecker
+
+class AddBarrierOpening(
+        private val checkArSupport: ArSupportChecker
+    ) : OsmElementQuestType<WidthAnswer> {
+
+    private val nodeFilter by lazy { """
+       nodes with
+         barrier ~ ${GATEWAYS.joinToString("|")}
+         and (!maxwidth:physical or source:maxwidth_physical ~ ".*estimat.*")
+         and (!width or source:width ~ ".*estimat.*")
+         and (!maxwidth or source:maxwidth ~ ".*estimat.*")
+         and access !~ private|no|customers|agricultural
+    """.toElementFilterExpression() }
+
+    private val excludedWaysFilter by lazy { """
+        ways with
+          highway and access ~ private|no
+    """.toElementFilterExpression() }
+
+    override val changesetComment = "Specify width of opening"
+    override val wikiLink = "Key:barrier"
+ // icon needed
+    override val icon = R.drawable.ic_quest_bicycleway_width
+    override val achievements = listOf(BICYCLIST, CAR, PEDESTRIAN, WHEELCHAIR)
```

Comment:
I'd suggest just wheelchair and bicyclist. Cars usually don't go through gates / gates on roads are broad enough for cars, of course. Gates that are not large enough for pedestrians wouldn't be gates, but uhh...

So anyway, please just use Wheelchair and maybe bicyclist.

Reply:
CAR & PEDESTRIAN removed from list.